### ISSUE TITLE
Add query batching

### ISSUE THREAD

**Issue by dljcollette** [SeMI-network/weaviate-graphql-prototype#34]
I'm unable to send batched queries to the prototype. Query batching is a powerful feature that allows to send an array of GraphQL query objects in the req.body instead of just one GraphQL query object. As a result, less requests need to be made and, more important, the client is able to dynamically compose multiple queries into one - a feature that comes in very handy with a dynamic app like Playground.

@laura-ham could you implement query batching so that I can send an array of query objects?

**Comment by bobvanluijt** [comment]
Hi @dljcollette, 

We are using data loader in Weaviate itself (https://github.com/creativesoftwarefdn/weaviate/tree/feature/graphql-dataloader) but I'm not sure how much it would add to also do this in the prototype.

Did I understand correctly that in Appolo is a simple `shouldBatch: true` vs `shouldBatch: false`?

**Comment by dljcollette** [comment]
Hi @bobvanluijt, the terminology is a bit confusing as there are two "batches":

1. Sending batched GraphQL operations in one HTTP request from client to server (in Apollo Client 1 it was indeed a matter of shouldBatch: boolean; the latest Apollo Client requires a so-called [Link](https://github.com/apollographql/apollo-link/tree/master/packages/apollo-link-batch-http)). The server should be able to accept a query of GraphQL queries if you're using batched queries, this is currently not the case.
2. Batching when a GraphQL operation is executed on the server (this is where where you could use dataloader).

I can create Playground without 2 but I won't be able to easily make it dynamic without 1.

**Comment by JeroenStravers1** [comment]
I don't think so. I've been looking through Weaviate's graphql request [entrypoint](https://github.com/creativesoftwarefdn/weaviate/blob/develop/restapi/operations/graphql/weaviate_graphql_post.go), and the struct used for the params of a request currently only allows for a single [HTTPRequest](https://github.com/creativesoftwarefdn/weaviate/blob/1d4b50f8f2a9fb47615c0c29da6540de1f386a0e/restapi/operations/graphql/weaviate_graphql_post_parameters.go#L45). The same could be true for regular http requests, I'd have to examine [configure_weaviate.go](https://github.com/creativesoftwarefdn/weaviate/blob/develop/restapi/configure_weaviate.go) further.

I've opened an [issue](https://github.com/creativesoftwarefdn/weaviate/issues/498) in Weaviate for this.
